hawkular-inventory-itest/src/test/java/org/hawkular/inventory/rest/test/InventoryITest.java

```diff
@@ -1288,69 +1301,69 @@ public void testComplexBulkCreate() throws Throwable {
-        delete(basePath + "/metricTypes/" + mt1);
+        delete(basePath + "/entity/mp;" + mpId);
+        delete(basePath + "/entity/e;" + env1);
+        delete(basePath + "/entity/e;" + env2);
+        delete(basePath + "/entity/rt;" + rt1);
+        delete(basePath + "/entity/mt;" + mt1);
     }
 
     @Test
     public void testMetadataPacks() throws Throwable {
-        Response response = post(basePath + "/metadatapacks",
+        Response response = post(basePath + "/entity/metadataPack",
                 "{ \"members\": [\"/t;" + tenantId + "/rt;" + urlTypeId + "\"]}");
         JsonNode json = mapper.readTree(response.body().string());
         String mpId = json.get("id").asText();
-        String url = baseURI + basePath + "/resourceTypes/" + urlTypeId;
+        String url = baseURI + basePath + "/entity/rt;" + urlTypeId;
         response = client.newCall(newAuthRequest().url(url).delete().build()).execute();
         assertEquals("Deleting a resource type that is part of metadatapack should not be possible.",
                 400, response.code());
 
-        delete(basePath + "/metadatapacks/" + mpId);
+        delete(basePath + "/entity/mp;" + mpId);
     }
 
     @Test
     public void testRecursiveChildren() throws Throwable {
         try {
-            Response response = post(basePath + "/" + environmentId +"/resources",
+            Response response = post(basePath + "/entity/e;" + environmentId +"/resource",
                 "{ \"id\": \"rootResource\", \"resourceTypePath\": \"/" + urlTypeId +"\"}");
             assertEquals(201, response.code());
 
-            response = post(basePath + "/" + environmentId +"/resources/rootResource",
+            response = post(basePath + "/entity/e;" + environmentId +"/r;rootResource/resource",
                 "{\"id\": \"childResource\", \"resourceTypePath\": \"/" + urlTypeId +"\"}" );
             assertEquals(201, response.code());
 
-            response = post(basePath + "/" + environmentId +"/resources/rootResource/childResource",
+            response = post(basePath + "/entity/e;" + environmentId +"/r;rootResource/r;childResource/resource",
                         "{\"id\": \"grandChildResource\", \"resourceTypePath\": \"/" + urlTypeId +"\"}");
             assertEquals(201, response.code());
 
-            response = post(basePath + "/" + environmentId +"/resources/rootResource/childResource",
+            response = post(basePath + "/entity/e;" + environmentId +"/r;rootResource/r;childResource/resource",
                         "{\"id\": \"grandChildResource2\", \"resourceTypePath\": \"/" + roomRTypeId + "\"}");
             assertEquals(201, response.code());
 
-            response = get(basePath + "/" + environmentId +"/resources/rootResource/recursiveChildren",
-                    "typeId", urlTypeId);
+            response = get(basePath + "/traversal/e;" + environmentId +"/r;rootResource/recursive/definedBy=%2Frt%3B"
```

Comment:
The trick here is that the canonical path is **already** percent-encoded and safe to use as part of URL. The `/traversal` endpoint is primarily meant for more complicated traversals. If you know a canonical path of something (and you really should only use that for identifying something, because "environmentId" is not enough to address the environment), you can use
```javascript
url = "/hawkular/inventory/entity" + canonicalPath
```
and that's it. No further encoding needed.